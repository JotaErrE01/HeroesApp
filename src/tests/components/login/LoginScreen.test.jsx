import { render, screen } from '@testing-library/react';
import { Router } from 'react-router';
import AuthContext from '../../../auth/AuthContext';
import LoginScreen from '../../../components/login/LoginScreen';
import { createMemoryHistory } from 'history';
import userEvent from '@testing-library/user-event';

describe('Pruebas en el componente <LoginScreen />', () => {

    Storage.prototype.getItem = jest.fn();
    const contextValue = {
        dispatch: jest.fn()
    }

    test('Debe de hacer match con el snapshot', () => {
        
        const { container } = render(
            <AuthContext.Provider value={contextValue}>
                <LoginScreen />
            </AuthContext.Provider>
        );

        expect( container ).toMatchSnapshot();
    })
    
    test('Debe de realizar el dispatch y la navegacion', () => {

        const history = {
            replace: jest.fn()
        }
        
        const { container } = render(
            <AuthContext.Provider value={contextValue}>
                <LoginScreen history={history}/>
            </AuthContext.Provider>
        );

        userEvent.click(container.querySelector('.btn-primary'));

        expect( contextValue.dispatch ).toHaveBeenCalled();
        expect( history.replace ).toHaveBeenCalled();
        expect( localStorage.getItem ).toHaveBeenCalledWith( 'lastPath' );

    })
    

})
