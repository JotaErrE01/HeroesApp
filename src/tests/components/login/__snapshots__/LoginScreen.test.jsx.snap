// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <LoginScreen /> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="container mt-5"
  >
    <h1>
      Login
    </h1>
    <hr />
    <button
      class="btn btn-primary"
    >
      Login
    </button>
  </div>
</div>
`;
